mounts/org.osbuild.ostree.deployment

```diff
@@ -35,7 +35,26 @@ SCHEMA_2 = """
       "deployment": {
         "type": "object",
         "additionalProperties": false,
-        "required": ["osname", "ref"],
+        "oneOf": [
+          {
+            "properties": {
+              "default": {"enum": [false]}
+            },
+            "required": ["osname", "ref"]
+          },
+          {
+            "properties": {
+              "default": {"enum": [true]}
+            },
+            "not": {
+              "anyOf": [
+                {"required": ["osname"]},
+                {"required": ["ref"]},
+                {"required": ["serial"]}
+              ]
+            }
+          }
+        ],
```

Comment:
This schema looks complicated, but all we're really saying here is:

> if you specify `default: true` then it's invalid to specify any of the other opions

which will help validate things early and reduce confusion.

If there is a more succinct way to describe that in the schema, let us know.